src/rebar_prv_shell.erl

```diff
@@ -328,7 +328,8 @@ reread_config(State) ->
         ConfigList ->
             try
                 [application:set_env(Application, Key, Val)
-                  || {Application, Items} <- ConfigList,
+                  || Config <- ConfigList,
+                     {Application, Items} <- Config,
```

Comment:
I believe this pattern now eliminates `{Application, Items}` when it is set at the top-level directly, since it will match into `Config`, and then refuse to match into `{Application, Items}` ?